app/src/main/java/com/ncnf/main/MainActivity.java

```diff
@@ -17,37 +20,76 @@
+
+        fragmentManager.beginTransaction().show(activeFragment).commit();
+
         navigationBar = findViewById(R.id.mainNavigationBar);
 
-        navigationBar.setOnNavigationItemSelectedListener(item -> {
-            if(item.getItemId() == R.id.navigation_feed){
-                getSupportFragmentManager().beginTransaction()
-                        .replace(R.id.mainFragmentContainerView, new FeedFragment())
-                        .commit();
+        navigationBar.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
+    }
+
 
-            } else if(item.getItemId() == R.id.navigation_map) {
-                getSupportFragmentManager().beginTransaction()
-                        .replace(R.id.mainFragmentContainerView, new MapFragment())
-                        .commit();
+    public final  BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener = item -> {
+        switch (item.getItemId()){
+            case R.id.navigation_home:
+                fragmentManager.beginTransaction().hide(activeFragment).show(homeFragment).commit();
+                activeFragment = homeFragment;
+                return true;
 
-            } else if(item.getItemId() == R.id.navigation_home){
-                getSupportFragmentManager().beginTransaction()
-                        .replace(R.id.mainFragmentContainerView, new HomeFragment())
-                        .commit();                }
+            case R.id.navigation_map:
+                fragmentManager.beginTransaction().hide(activeFragment).show(mapFragment).commit();
+                activeFragment = mapFragment;
+                return true;
+
+            case R.id.navigation_feed:
+                fragmentManager.beginTransaction().hide(activeFragment).show(feedFragment).commit();
+                activeFragment = feedFragment;
+                return true;
+        }
+
+        return false;
+    };
+
+    @Override
+    protected void onSaveInstanceState(@NonNull Bundle outState) {
+        super.onSaveInstanceState(outState);
+
+        fragmentManager.putFragment(outState, "home_fragment", homeFragment);
+        fragmentManager.putFragment(outState, "map_fragment", mapFragment);
+        fragmentManager.putFragment(outState, "feed_fragment", feedFragment);
+        fragmentManager.putFragment(outState, "active_fragment", activeFragment);
+    }
+
+    @Override
+    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
+        super.onRestoreInstanceState(savedInstanceState);
 
-            return true;
-        });
+        this.homeFragment = fragmentManager.getFragment(savedInstanceState, "home_fragment");
+        this.mapFragment = fragmentManager.getFragment(savedInstanceState, "map_fragment");
+        this.feedFragment = fragmentManager.getFragment(savedInstanceState, "feed_fragment");
+        this.activeFragment = fragmentManager.getFragment(savedInstanceState, "active_fragment");
```

Comment:
Should create final strings as class attributes for the string values

Reply:
Done !